go/libraries/doltcore/doltdb/doltdb.go

```diff
@@ -573,9 +591,15 @@ func (ddb *DoltDB) CommitDanglingWithParentCommits(ctx context.Context, valHash
 		}
 		parents = append(parents, addr)
 	}
-
 	commitOpts := datas.CommitOptions{Parents: parents, Meta: cm}
-	dcommit, err := datas.NewCommitForValue(ctx, datas.ChunkStoreFromDatabase(ddb.db), ddb.vrw, ddb.ns, val, commitOpts)
+
+	return ddb.CommitDangling(ctx, val, commitOpts)
+}
+
+func (ddb *DoltDB) CommitDangling(ctx context.Context, val types.Value, opts datas.CommitOptions) (*Commit, error) {
```

Comment:
Exported methods should have comments.  In a different exported method there is this comment:

// dangling commits are unreferenced by any branch or ref. They are created in the course of programmatic updates
// such as rebase. You must create a ref to a dangling commit for it to be reachable

Obviously don't use that but telling a dev what a dangling commit is will save some confusion.